src/scanner.c

```diff
@@ -1275,10 +1290,27 @@ static inline void GET_CHAR( void ) {
 }
 
 static inline void UNGET_CHAR( Char c ) {
-  assert(STATE(In) != &Pushback);
-  Pushback = c;
-  RealIn = STATE(In);
-  STATE(In) = &Pushback;
+    assert(!CHAR_PUSHBACK_EMPTY());
+    Pushback = c;
+    RealIn = STATE(In);
+    STATE(In) = &Pushback;
+}
+
+// Get current line position. In the case where we pushed back the last
+// character on the previous line we return the first character of the
+// current line, as cannot retrieve the previous line.
```

Comment:
Missing "we" after "as"?

Reply:
Fixed (along with some other minor issues which were causing tests to fail)